### ISSUE TITLE
Make sure codalab service runs smoothly for a new user

### ISSUE THREAD

**Issue by HidyHan** [codalab/codalab-worksheets#4567]
I had various issues trying to run codalab on my machine, including
- Docker has no permission to mount on my codalab code dir.
- Docker has no permission to create files (config.json, state.json, etc.) in desired locations.
- Synthesized file path seems malformed (e.g. double // in the file path).

We should make sure codalab service runs smoothly, by either adding documentation, or making sure the desired user is used in the startup file (might be difficult if the service is not started with sudo).

**Comment by HidyHan** [comment]
My /var/lib/mysql and /var/log/mysql directory's uid and gid are both mysql, however $CODALAB_UID is 1000:1000 which is set [here](https://github.com/codalab/codalab-worksheets/blob/master/codalab_service.py#L215). I manually changed the user field under mysql rule in docker_compose_custom.yml to mysql:mysql, but still see the permission error in creating the mysql container.

Also confusingly mysql container creation error won't show up at startup but only later.

**Comment by HidyHan** [comment]
Platform: ubuntu 22.04 jammy

Things I tried (in addition to just running `./codalab_service.py start -bd`):

I did post installation setup: https://docs.docker.com/engine/install/linux-postinstall/.

I downgraded my docker version to 20.10.13 from the latest. Andrew said he ran into a similar issue a year ago with the then latest version and also downgraded. Docker may have changed their bind mount behavior which doesn't play nicely with the existing codalab code.

I ran into file permission issue [here](https://github.com/codalab/codalab-worksheets/blob/984573d57c01cee083bd9a9e51186f7a712a698e/codalab/lib/codalab_manager.py#L218) when writing the config.json file. I did os.getuid() and os.getgid() before the write and verified the permission of the write location is given to the correct uid and gid, but still the permission issue happens. I did `sudo chmod -R 777 PATH_TO_CODE/codalab-worksheets/var/` to get over this in a janky way.

I ran into `process: ERROR: for nginx  Cannot start service nginx: Ports are not available: exposing port TCP 0.0.0.0:80 -> 0.0.0.0:0: listen tcp 0.0.0.0:80: bind: permission denied`. I changed the [port number](https://github.com/codalab/codalab-worksheets/blob/984573d57c01cee083bd9a9e51186f7a712a698e/codalab_service.py#L242) to be 8080. I verified that no other process was running on port 80, so no idea why the error occurred in the first place. The nginx docker user is 0:0, so there shouldn't be permission issues.

At last I was able to bring up the service, but running the actual bundle triggers `2023-11-12 17:28:40 rest-server_1     | OSError: AF_UNIX path too long`. Since the service can start up ok and this is due to my local path configuration, closing this issue, but it was a bumpy journey!
